### DISCUSSION TITLE
Invalid dataprovider

### DISCUSSION THREAD

**Discussion by m-kuhn** [opengisch/QField#2452] [Q&A]
I try to setting my project on QFieldCluod, and for each layer that I created on myself, I get this kind of error "Exported layer 'layer_name' is not valid: 'invalid_dataprovider'". Can someone help me?

Open question by @RubenSadei migrated from https://github.com/opengisch/qfieldcloud-beta/discussions/8

**Comment by suricactus** [comment]
Hey, @RubenSadei. Thanks for testing QFieldCloud. Can you give more details about your layers? I guess they are some kind of online provider  (PostGIS, WFC etc) that require login credentials OR point to a host that is not available from QFieldCloud (`localhost`, 0.0.0.0, etc). Ideally you can share your project with us.

**Reply by niall-quinlan** [reply]
I think I am having the same issue - I posted about it here: [https://github.com/opengisch/qfieldcloud-beta/discussions/1#discussioncomment-803842](url). From your reply about, I think I misunderstood that the Postgres database needs to be hosted publicly.

**Reply by suricactus** [reply]
Hi, @niall-quinlan . Sorry I missed your comment there. Yes, Postgres database should be publicly available in order to prepare package your layers on QFieldCloud and make you project available on any device.

**Reply by RubenSadei** [reply]
@suricactus Thanks, how can I share my project? In my project I've online provider, but these don't give me troubles!

**Reply by RubenSadei** [reply]
I fixed the problem, but not in the best way. If I move all shapefiles to the same folder as the project then it no longer gives me this error and everything works. But from a practical point of view, in keeping the files in order, it is quite problematic. 

I've tried loading an online layer, for example the one from google sattelite, it partially loads the image. I think this is also a bug.

**Reply by suricactus** [reply]
Hi, RubenSadei.

Yes, QFieldCloud assumes all your files are in the same directory. There is an upcomming functionality for QFieldSync to collect all your files in the same folder, so you don't need to do this manually.

On the Google Satellite problem, can you elaborate more? What do you mean by "partially loads the image"? What is the image - a local raster file, WMS, XYZ layer? Did you mark it also as base layer? BTW, feel free to open a new question in discussion to resolve your Google Satellite problem.

**Comment by marioba** [comment]
In order to provide a clearer and more manageable discussion forum and to have all discussions about QField and related products (i.e. QFieldCloud and QFieldSync) in a single place, we transferred all the discussions from the QFieldCloud to the [QField repository](https://github.com/opengisch/QField/discussions).

**Comment by Nioigte** [comment]


**Reply by suricactus** [reply]
Opening a new issue/discussion, describing your project and data (data providers and files you have), pasting the QFieldCloud feedback text and screenshot, uploading an example project (if possible) and using less capital letters will drastically increase the chances someone helps you. ;)

**Comment by shafibaj07** [comment]
I also had this problem with my Shapefiles (local files, no database involved) . The trick is to choose the layer that has the issue and edit it in QGIS. By editing I mean, **simply add some attributes to one of the features or digitize a feature**. After you finished the editing, try Uploading it to QFieldCloud or Synchronize it. It should work without issue now . You can delete the edits you made earlier via QGIS or Qfield app if you want.
